src/Microsoft.Azure.WebJobs.ServiceBus/EventHubs/EventHubTriggerInputBindingStrategy.cs

```diff
@@ -42,72 +42,80 @@ public EventData[] BindMultiple(EventHubTriggerInput value, ValueBindingContext
             return value.Events;
         }
 
-        // Get the static binding contract
-        //  - gets augmented 
-        public Dictionary<string, Type> GetStaticBindingContract()
+        public Dictionary<string, Type> GetBindingContract(bool isSingleDispatch = true)
         {
-            // TODO: need to be able to determine here if we're single dispatch
-            // https://github.com/Azure/azure-webjobs-sdk/issues/1072
-            bool isSingleDispatch = true;
-
-            return GetBindingContract(isSingleDispatch);
-        }
-
-        internal static Dictionary<string, Type> GetBindingContract(bool isSingleDispatch)
-        {
-            Dictionary<string, Type> contract = new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase);
+            var contract = new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase);
             contract.Add("PartitionContext", typeof(PartitionContext));
 
-            // TODO: need to be able to determine here if we're single dispatch
-            // https://github.com/Azure/azure-webjobs-sdk/issues/1072
-            if (isSingleDispatch)
-            {
-                AddBindingContractMember(contract, "PartitionKey", typeof(string), isSingleDispatch);
-                AddBindingContractMember(contract, "Offset", typeof(string), isSingleDispatch);
-                AddBindingContractMember(contract, "SequenceNumber", typeof(long), isSingleDispatch);
-                AddBindingContractMember(contract, "EnqueuedTimeUtc", typeof(DateTime), isSingleDispatch);
-                AddBindingContractMember(contract, "Properties", typeof(IDictionary<string, object>), isSingleDispatch);
-                AddBindingContractMember(contract, "SystemProperties", typeof(IDictionary<string, object>), isSingleDispatch);
-            }
+            AddBindingContractMember(contract, "PartitionKey", typeof(string), isSingleDispatch);
+            AddBindingContractMember(contract, "Offset", typeof(string), isSingleDispatch);
+            AddBindingContractMember(contract, "SequenceNumber", typeof(long), isSingleDispatch);
+            AddBindingContractMember(contract, "EnqueuedTimeUtc", typeof(DateTime), isSingleDispatch);
+            AddBindingContractMember(contract, "Properties", typeof(IDictionary<string, object>), isSingleDispatch);
+            AddBindingContractMember(contract, "SystemProperties", typeof(IDictionary<string, object>), isSingleDispatch);
 
             return contract;
         }
 
         private static void AddBindingContractMember(Dictionary<string, Type> contract, string name, Type type, bool isSingleDispatch)
         {
+            if (!isSingleDispatch)
+            {
+                name += "Array";
```

Comment:
Chose a naming pattern for array based binding data. If other bindings do this in the future, we'd enforce this pattern, so we need to decide on a good name. Rather than try to apply some pluralization heuristic (always error prone), I just append "Array".